src/Compilers/CSharp/Test/Semantic/Diagnostics/OperationAnalyzerTests.cs

```diff
@@ -1145,6 +1145,27 @@ public void M1()
         M0(1, new int[] { 2, 3, 4 });
         M0(1, new int[] { 2, 3, 4, 5 });
         M0(1, new int[] { 2, 3, 4, 5, 6 });
+        M2(1, c: 2);
+        D d = new D(3, c: 40);
+        d = new D(""Hello"", 1, 2, 3, 4);
+        d = new D(""Hello"", new int[] { 1, 2, 3, 4 });
+        d = new D(""Hello"", 1, 2, 3);
+        d = new D(""Hello"", new int[] { 1, 2, 3 });
```

Comment:
What happens if we do something like this?
```cs
var array = new int[] { 1, 2, 3, 4 }
d = new D("Hello", array );
```

Will `array` be treated as a params array argument?